scripts/build_test.py

```diff
@@ -14,29 +14,313 @@
+
+    def test_verify_filepath_hash(self):
+        """Test _verify_filepath_hash raises exception:
+            1) When there is an empty hash dict.
+            2) When a filename is expected to contain hash but does not.
+            3) When there is a hash in filename that cannot be found in
+                hash dict.
+        """
+        # Final filepath example: base.240933e7564bd72a4dde42ee23260c5f.html.
+        file_hashes = dict()
+        base_filename = 'base.html'
+        with self.assertRaisesRegexp(ValueError, 'Hash dict is empty'):
+            build._verify_filepath_hash(base_filename, file_hashes)
+
+        # Generate a random hash dict for base.html.
+        file_hashes = {base_filename: random.getrandbits(128)}
+        with self.assertRaisesRegexp(
+            ValueError, '%s is expected to contain MD5 hash' % base_filename):
+            build._verify_filepath_hash(base_filename, file_hashes)
+
+        bad_filepath = 'README'
+        with self.assertRaisesRegexp(
+            ValueError, 'Filepath has less than 2 partitions after splitting'):
+            build._verify_filepath_hash(bad_filepath, file_hashes)
+
+        hashed_base_filename = build._insert_hash(
+            base_filename, random.getrandbits(128))
+        with self.assertRaisesRegexp(
+            KeyError,
+            'Hash from file named %s does not match hash dict values' %
+            hashed_base_filename):
+            build._verify_filepath_hash(hashed_base_filename, file_hashes)
+
+    def test_process_html(self):
+        """Test process_html removes whitespaces and adds hash to filepaths."""
+        BASE_HTML_SOURCE_PATH = os.path.join(TEMPLATES_DEV_DIR, 'base.html')
+        BASE_JS_SOURCE_PATH = os.path.join(
+            TEMPLATES_DEV_DIR, 'pages', 'Base.js')
+        build._ensure_files_exist([BASE_HTML_SOURCE_PATH, BASE_JS_SOURCE_PATH])
+        # Prepare a file_stream object from StringIO.
+        minified_html_file_stream = StringIO.StringIO()
+        # Obtain actual file hashes of /templates to add hash to all filepaths
+        # within the HTML file. The end result will look like:
+        # E.g <script ... app.js></script>
+        # --> <script ... app.[hash].js></script>.
+        # Only need to hash Base.js.
+        with self.swap(build, 'FILE_EXTENSIONS_TO_IGNORE', ('.html',)):
+            file_hashes = build.get_file_hashes(TEMPLATES_DEV_DIR)
+
+        # Assert that base.html has white spaces and has original filepaths.
+        with open(BASE_HTML_SOURCE_PATH, 'r') as source_base_file:
+            source_base_file_content = source_base_file.read()
+            self.assertRegexpMatches(
+                source_base_file_content, r'\s{2,}',
+                msg="No white spaces detected in %s unexpectedly"
+                % BASE_HTML_SOURCE_PATH)
+            for filepath in file_hashes:
+                # Looking for templates/pages/Base.js in
```

Comment:
Why not just look for it directly?

self.assertIn('templates/pages/Base.js', source_base_file_content)

The more straightforward the tests are, the better.